androidApp/src/main/java/dev/johnoreilly/confetti/ui/ConfettiApp.kt

```diff
@@ -24,80 +24,151 @@ import androidx.compose.material3.Scaffold
+import org.koin.androidx.compose.get
 
-@OptIn(
-    ExperimentalMaterial3Api::class,
-    ExperimentalLayoutApi::class,
-)
 @Composable
 fun ConfettiApp(
-    navController: NavHostController = rememberNavController(),
+    navController: NavHostController,
     windowSizeClass: WindowSizeClass,
     displayFeatures: List<DisplayFeature>,
-    appState: ConfettiAppState = rememberConfettiAppState(windowSizeClass, displayFeatures, navController)
 ) {
-    ConfettiTheme {
-        ConfettiBackground {
-            Scaffold(
-                containerColor = Color.Transparent,
-                contentColor = MaterialTheme.colorScheme.onBackground,
-                contentWindowInsets = WindowInsets(0, 0, 0, 0),
-                bottomBar = {
-                    if (appState.shouldShowBottomBar) {
-                        ConfettiBottomBar(
-                            destinations = appState.topLevelDestinations,
-                            onNavigateToDestination = appState::navigate,
-                            currentDestination = appState.currentDestination
-                        )
-                    }
-                }
-            ) { padding ->
-                Row(
-                    Modifier
-                        .fillMaxSize()
-                        .windowInsetsPadding(
-                            WindowInsets.safeDrawing.only(
-                                WindowInsetsSides.Horizontal
-                            )
-                        )
-                ) {
-                    if (appState.shouldShowNavRail) {
-                        ConfettiNavRail(
-                            destinations = appState.topLevelDestinations,
-                            onNavigateToDestination = appState::navigate,
-                            currentDestination = appState.currentDestination,
-                            modifier = Modifier.safeDrawingPadding()
-                        )
-                    }
+    val appState: ConfettiAppState = rememberConfettiAppState(
+        windowSizeClass,
+        displayFeatures,
+        navController
+    )
 
-                    ConfettiNavHost(
-                        navController = appState.navController,
-                        isExpandedScreen = appState.isExpandedScreen,
-                        displayFeatures = appState.displayFeatures,
-                        onBackClick = appState::onBackClick,
-                        onNavigateToDestination = appState::navigate,
-                        modifier = Modifier
-                            .padding(padding)
-                            .consumeWindowInsets(padding)
+    val repository: ConfettiRepository = get()
+    val conference by repository.getConferenceFlow()
+        .collectAsState(initial = null)
+
+    if (conference == null) {
+        // Reading from the settings
+        CircularProgressIndicator()
+    } else {
+        val initialRoute = if (conference == CONFERENCE_NOT_SET) {
```

Comment:
I'll try to dig it out, but I'm pretty sure I've seen advice to not change the startDestination in this case. But if it's working, let's go with it until I can find that and it's worth changing.

Reply:
Yea, it's recomposing the whole `NavHost`... Feels a bit weird but on the other hand, it's almost as if we restarted the app. I'm curious if you find anything.